frame/examples/offchain-worker-ping-pong/src/lib.rs

```diff
@@ -0,0 +1,681 @@
+				// signature it only checks if the signature is coherent with
+				// the public key of `SignedPayload` whoever that might be (not
+				// necessarily an authorized OCW)
+				let signature_valid =
+					SignedPayload::<T>::verify::<T::AuthorityId>(payload, signature.clone());
+				if !signature_valid {
+					return InvalidTransaction::BadProof.into()
+				}
+				Self::validate_transaction_parameters(&payload.block_number)
+			} else if let Call::pong_unsigned { block_number, nonce: _n } = call {
+				Self::validate_transaction_parameters(block_number)
+			} else {
+				InvalidTransaction::Call.into()
+			}
+		}
+	}
+
+	/// Container for different types that implement [`DefaultConfig`]` of this pallet.
+	pub mod config_preludes {
+		// This will help use not need to disambiguate anything when using `derive_impl`.
+		use super::*;
+
+		/// A type providing default configurations for this pallet in testing environment.
+		pub struct TestDefaultConfig;
+		const UNSIGNED_PRIORITY: u64 = 1 << 20;
+
+		#[frame_support::register_default_impl(TestDefaultConfig)]
+		impl DefaultConfig for TestDefaultConfig {
+			type UnsignedPriority = frame_support::traits::ConstU64<UNSIGNED_PRIORITY>;
+			type MaxPings = frame_support::traits::ConstU32<64>;
+			type MaxAuthorities = frame_support::traits::ConstU32<64>;
+		}
+	}
+}
+
+/// Payload used by this example crate to hold pong response data required to
+/// submit an unsigned transaction.
+#[derive(Encode, Decode, Clone, PartialEq, Eq, RuntimeDebug, scale_info::TypeInfo)]
+pub struct PongPayload<Public, BlockNumber> {
+	block_number: BlockNumber,
+	nonce: u32,
+	public: Public,
+}
+
+impl<T: SigningTypes> SignedPayload<T> for PongPayload<T::Public, BlockNumberFor<T>> {
+	fn public(&self) -> T::Public {
+		self.public.clone()
+	}
+}
+
+impl<T: Config> Pallet<T> {
+	fn is_authority(who: &T::AccountId) -> bool {
+		<Authorities<T>>::get().contains(who)
+	}
+
+	fn validate_transaction_parameters(block_number: &BlockNumberFor<T>) -> TransactionValidity {
+		// Now let's check if the transaction has any chance to succeed.
+		let next_unsigned_at = <NextUnsignedAt<T>>::get();
+		if &next_unsigned_at > block_number {
+			return InvalidTransaction::Stale.into()
+		}
+		// Let's make sure to reject transactions from the future.
+		let current_block = <system::Pallet<T>>::block_number();
+		if &current_block < block_number {
+			return InvalidTransaction::Future.into()
+		}
+
+		ValidTransaction::with_tag_prefix("ExampleOffchainWorker")
+			// We set base priority to 2**20 and hope it's included before any
+			// other transactions in the pool.
+			.priority(T::UnsignedPriority::get().saturating_add(0))
+			// This transaction does not require anything else to go before into
+			// the pool. In theory we could require `previous_unsigned_at`
+			// transaction to go first, but it's not necessary in our case.
+			//.and_requires() We set the `provides` tag to be the same as
```

Comment:
Something here looks off.

Reply:
this is coming from the original example:
https://github.com/paritytech/substrate/blob/28b1f79abedbad2ec97b49277b36e1f797ca9e5d/frame/examples/offchain-worker/src/lib.rs#L705-L713

what do you feel might be wrong, specifically?